homeassistant/components/feedreader.py

```diff
@@ -103,8 +120,51 @@ def _publish_new_entries(self):
         self._firstrun = False
 
 
+class StoredData(object):
+    """Abstraction over pickle data storage."""
+
+    def __init__(self, data_file):
+        """Initialize pickle data storage."""
+        self._data_file = data_file
+        self.lock = Lock()
+
+    def _get_data(self):
+        """Fetch data stored into pickle file."""
+        if exists(self._data_file):
```

Comment:
To prevent a lot of read/writes every update, it would be better if `StoredData` caches the file and invalidates the cache whenever `put_timestamp` is called.